resources/js/humhub.firebase.js

```diff
@@ -22,21 +33,49 @@ humhub.module('firebase', function (module, require, $) {
         }
     };
 
+    /**
+     * Gets the content to display for notification permission status.
+     * @function
+     * @returns {string} The message corresponding to the current notification permission status.
+     */
+    const getNotificationPermissionContent = function () {
+        if (!("Notification" in window)) {
+            return module.text('status.not-supported');
+        }
+        switch (Notification.permission) {
+            case "granted":
+                return module.text('status.granted');
+            case "denied":
+                return module.text('status.denied');
+            default:
+                return module.text('status.default');
+        }
+    }
+
+    /**
+     * Adds information about push notification permissions to the UI.
+     * @function
+     */
+    const addPushNotificationPermissionsInfo = function () {
+        let content = getNotificationPermissionContent();
+        $('#pushNotificationPermissionInfo').html(content);
+    }
+
+    /**
+     * Handles actions after the service worker registration.
+     * @function
+     * @param {ServiceWorkerRegistration} registration - The service worker registration object.
+     */
     const afterServiceWorkerRegistration = function (registration) {
-        //console.log("After Service Worker Registration");
-        //console.log(registration);
-
         const that = this;
 
         this.messaging.useServiceWorker(registration);
 
-        // Request for permission
         this.messaging.requestPermission().then(function () {
-            //console.log('Notification permission granted.');
+            addPushNotificationPermissionsInfo('granted');
```

Comment:
The parameter here isn't also not longer used?

Reply:
I am also not sure if we need these calls `addPushNotificationPermissionsInfo` in the `afterServiceWorkerRegistration` at all.
Because usually the ServiceWorkerRegistration takes place very early, e.g. at login, and the PermissionInfo is not displayed anywhere here. In addition, the registration is only once.

Reply:
I'd have to look into this more when I have the time to do so. 🤔